IDEAS/Buildings/Components/RectangularZoneTemplate.mo

```diff
@@ -1281,6 +1281,12 @@ initial equation
   AWallD = lD*h - (if hasWinD then  propsBusInt[indWinD].area else 0);
   ACei = A - (if hasWinCei then  propsBusInt[indWinCei].area else 0);
 
+  // assert that the windows are not bigger than the wall
+  assert(AWallA >= 0, "The netto surface area of wall A is negative. This is not allowed.");
+  assert(AWallB >= 0, "The netto surface area of wall B is negative. This is not allowed.");
+  assert(AWallD >= 0, "The netto surface area of wall C is negative. This is not allowed.");
+  assert(AWallD >= 0, "The netto surface area of wall D is negative. This is not allowed.");
+  assert(ACei >= 0, "The netto surface area of the ceiling is negative. This is not allowed.");
```

Comment:
Surprisingly, when AWallA becomes negative, no assert is triggered. I only get in the simulation tab:
```
dymosim started
... "dsin.txt" loading (dymosim input file)
Error: The following error was detected at time: 0

Model error - sqrt (-2.3999999999999986) = sqrt (-2.4)

Error: Integrator failed to start model.
```

Moving the assert statement to the equation section did not help.

Reply:
This can be solve by using a max(0,value) on AWall and in assert, using not ==0.